src/field.rs

```diff
@@ -190,6 +196,72 @@ pub trait FieldElement:
         }
         Ok(vec)
     }
+
+    /// Encode `input` as `bits`-bit vector of elements of `Self` if it's small enough
+    /// to be represented with that many bits.
+    ///
+    /// # Arguments
+    ///
+    /// * `input` - The field element to encode
+    /// * `bits` - The number of bits to use for the encoding
+    fn encode_into_bitvector_representation(
+        input: &Self::Integer,
+        bits: usize,
+    ) -> Result<Vec<Self>, FieldError> {
+        // Create a mutable copy of `input`.
+        // In each iteration of the following loop
+        // we take the least significant bit,
+        // and shift input to the right by one bit.
```

Comment:
nit: Block comments should break at 100 characters
```suggestion
        // Create a mutable copy of `input`. In each iteration of the following loop we take the
        // least significant bit, and shift input to the right by one bit.
```